feat(feedback): Add UF screenshot capability (#9867)
getsentry/sentry-docs: docs/platforms/javascript/common/user-feedback/index.mdx

To set up the integration, add the following to your Sentry initialization. There are many options you can pass to the integration constructor. See the [configuration documentation](/platforms/javascript/user-feedback/configuration/) for more details.

<SignInNote />

<PlatformContent includePath="user-feedback/setup" />

By default, this will insert the widget into the bottom right corner of your website. You're free to customize nearly every aspect of the widget, including replacing it completely with your own UI.

> On SDK v8+, users can send screenshots with their feedback by default. You can configure this using the `showScreenshot` option. Screenshots aren't supported on mobile devices, so the screenshot button will be hidden in this case.

Reviewer: ```suggestion
On SDK v8 and above, users can send screenshots with their feedback by default. You can configure this using the `showScreenshot` option. Screenshots aren't supported on mobile devices, so the screenshot button will be hidden in this case.
```

> On SDK v8 and above, users can send screenshots with their feedback by default. You can configure this using the `showScreenshot` option. Screenshots aren't supported on mobile devices, so the screenshot button will be hidden in this case.

Reviewer 2: ```suggestion
On SDK v8 and above, users can send screenshots with their feedback. You can configure this using the `showScreenshot` option, by default it is set to `true`. Screenshots aren't supported on mobile devices, so the screenshot button will be hidden automatically in this case.
```

> Users will be able to send screenshots with their feedback by default. You can configure this using the `showScreenshot` option. Screenshots aren't supported on mobile devices, so the screenshot button will be hidden in this case. This feature is only available on SDK v8.

Reviewer 3: ```suggestion
On SDK v8+, users can send screenshots with their feedback by default. You can configure this using the `showScreenshot` option. Screenshots aren't supported on mobile devices, so the screenshot button will be hidden in this case.
```

> On SDK v8 and above, users can send screenshots with their feedback. You can configure this using the `showScreenshot` option, by default it is set to `true`. Screenshots aren't supported on mobile devices, so the screenshot button will be hidden automatically in this case.

Reviewer 4: We internally refer to it as `v8` but I think calling the exact version explicitly will be less confusing:

```suggestion
On SDK version 8.0.0 and above, users can send screenshots with their feedback. You can configure this using the `showScreenshot` option, by default it is set to `true`. Screenshots aren't supported on mobile devices, so the screenshot button will be hidden automatically in this case.
```

> On SDK version 8.0.0 and above, users can send screenshots with their feedback. You can configure this using the `showScreenshot` option, by default it is set to `true`. Screenshots aren't supported on mobile devices, so the screenshot button will be hidden automatically in this case.

Reviewer 5: can we mention here (and on the config page above too) the min. version for on-prem users to have that enables screenshot support?

Reviewer 5: We've got:
- SaaS -> Software as a service, sentry.io where people signup and login on their own. There's a single database will all the orgs and users and so on saved.
- ST -> Single Tenant, a separate deployment of sentry that we (the company) manage. We run the servers, but the servers only store information for one company.
- On-Prem -> Someone downloads sentry from github and runs it themselves. This'll also (almost always) just have 1 org saved into the database.

S4S is an example of ST where we can deploy the sentry codebase and not affect sentry.io. ST and On-Prem have very similar configs, while the SaaS codebase is special because it has the extra billing layer added on top (all the features are the same everywhere though)

Author: Updated! Feel free to make a suggestion if the wording is iffy

> On SDK version 8.0.0 and above, users can send screenshots with their feedback. If you're using on-prem, you also need release 24.4.2 and above. You can configure this using the `showScreenshot` option, by default it is set to `true`. Screenshots aren't supported on mobile devices, so the screenshot button will be hidden automatically in this case.

Reviewer 4: We call it Self-Hosted now: https://github.com/getsentry/self-hosted/

```suggestion
On SDK version 8.0.0 and above, users can send screenshots with their feedback. If you're self-hosting, you also need release 24.4.2 and above. You can configure this using the `showScreenshot` option, by default it is set to `true`. Screenshots aren't supported on mobile devices, so the screenshot button will be hidden automatically in this case.
```

<Alert level="">
  Screenshots use your [attachments quota](/product/accounts/quotas/manage-attachments-quota). All plans come with 1GB of attachments, which is approxiamately 2500 screenshots.
</Alert>

### Session Replay

> The User Feedback Widget integrates easily with <PlatformLink to="/session-replay/">Session Replay</PlatformLink>. First, make sure that the Session Replay integration is configured correctly and that `replaysOnErrorSampleRate` is greater than 0. When this is done, the Replay SDK will buffer up to 30 seconds worth of events until the user *opens* the User Feedback Widget. If the user submits feedback, you'll be able to view the Feedback (including the replay), inside of Sentry's web application.

Reviewer: ```suggestion
The User Feedback widget integrates easily with <PlatformLink to="/session-replay/">Session Replay</PlatformLink>. First, make sure that the Session Replay integration is configured correctly and that `replaysOnErrorSampleRate` is greater than 0. When this is done, the Replay SDK will buffer up to 30 seconds worth of events until the user *opens* the User Feedback widget. If the user submits feedback, you'll be able to view the feedback (including the replay), inside of Sentry's web application.
```

Reviewer 2: agree w above!

> The User Feedback widget integrates easily with <PlatformLink to="/session-replay/">Session Replay</PlatformLink>. First, make sure that the Session Replay integration is configured correctly and that `replaysOnErrorSampleRate` is greater than 0. When this is done, the Replay SDK will buffer up to 30 seconds worth of events until the user *opens* the User Feedback widget. If the user submits feedback, you'll be able to view the feedback (including the replay), inside of Sentry's web application.

Reviewer 4: We're not really buffering events. How about:

```suggestion
The User Feedback widget integrates easily with <PlatformLink to="/session-replay/">Session Replay</PlatformLink>. First, make sure that the Session Replay integration is configured correctly and that `replaysOnErrorSampleRate` is greater than 0. When this is done, the Replay SDK will buffer up to 30 seconds of the user's session until the user *opens* the User Feedback widget. If the user submits feedback, you'll be able to view the feedback (including the replay), inside of Sentry's web application.
```

> The User Feedback widget integrates easily with <PlatformLink to="/session-replay/">Session Replay</PlatformLink>. First, make sure that the Session Replay integration is configured correctly and that `replaysOnErrorSampleRate` is greater than 0. When this is done, the Replay SDK will buffer up to 30 seconds of the user's session until the user *opens* the User Feedback widget. If the user submits feedback, you'll be able to view the feedback (including the replay), inside of Sentry's web application.

Reviewer 4: >  inside of Sentry's web application.

@lizokm should we say "inside Sentry" ? I wonder if we have a consistent way to address this across the docs?

Reviewer 3: I think we can just say "in Sentry.io"
